lnrpc/routerrpc/config.go

```diff
@@ -0,0 +1,28 @@
+package routerrpc
+
+import (
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// RoutingConfig contains the configurable parameters that control routing.
+type RoutingConfig struct {
```

Comment:
It would be a circular dependency.

Reply:
Ah no, that is not the reason. There is no circular dependency.

The reason that it is here, is because it is a helper struct that only exists to deal with the conditional compilation and different variants of the sub server config struct. Without conditional compilation, we could have just taken those values from `routerrpc.Config`. 

I have expressed my opinion on conditional compilation multiple times before. In short, I advocate for dropping conditional compilation and clearly marking experimental rpc calls as experimental in the docs. 

This code and discussion is again something that for me counts towards unnecessary complications without upside.